Dashboard/Connectors/Serial/SerialConnector.cs

```diff
@@ -38,18 +49,32 @@ public void Stop()
         // Remove the event handler
         comPort.DataReceived -= OnDataReceived;
 
+        _heartbeatTimer.AutoReset = false;
+        _heartbeatTimer.Stop();
+        _heartbeatTimer.Dispose();
+
         // Close the serial port
         comPort.Close();
     }
 
+    public bool SkipedFirst = false;
+
     private void ParseMessage(string message)
     {
         // All messages start with the format "#ID=<ID>|UTC=<TIME>|<MSG>", so we split the message by the '|', and remove the first 2 elements
         // First we validate that we have the correct message format
         if (!message.StartsWith("#ID=") || !message.Contains("|UTC="))
             throw new InvalidOperationException("Invalid message format received", new Exception(message));
+        // Very dirty method right now, Need to ask Nick how to better handle the first message being out of format
```

Comment:
I fixed this stuff in ADL-45, this is a reminder for me to ensure this is fixed post merge (whichever ticket goes first)